mlir-tensorrt/executor/lib/Runtime/API/API.cpp

```diff
@@ -691,6 +729,19 @@ StatusOr<std::unique_ptr<MemRefValue>> MemRefValue::create(
     return getInvalidArgStatus("a specific device must be provided for MemRefs "
                                "that are device-visible");
 
+  // Check if given strides match canonical stride
+  if (assertCanonicalStrides && *assertCanonicalStrides) {
+    std::vector<int64_t> canonicalStride = getCanonicalStride(shape);
```

Comment:
It seems worthwhile to just combine the `getCanonicalStride` function and the `areStridesEquivalent` to avoid creating the `std::vector`, especially since speed was your concern for moving this check to C++.

Reply:
We want to print the expected canonical stride in the error message. I don't think we can avoid creating an extra vector copy.